exporter/elasticsearchexporter/model.go

```diff
@@ -474,9 +482,21 @@ func (m *encodeModel) encodeAttributesOTelMode(document *objmodel.Document, attr
 		}
 		return false
 	})
+	if stringifyArrayValues {
+		mapStringifyArrayValues(attributeMap)
+	}
 	document.AddAttributes("attributes", attributeMap)
 }
 
+func mapStringifyArrayValues(m pcommon.Map) {
+	m.Range(func(k string, v pcommon.Value) bool {
+		if v.Type() == pcommon.ValueTypeSlice {
```

Comment:
This not only affects array values but also other complex attribute values, like `ValueTypeMap`. Not quite sure how to deal with `ValueTypeBytes` tbh. How do we serialize it normally? Is that compatible with a field type that [supports dimensions](https://www.elastic.co/guide/en/elasticsearch/reference/current/tsds.html#time-series-dimension)?

Reply:
according to https://opentelemetry.io/docs/specs/otel/common/ 
> The attribute value is either:
>
>    A primitive type: string, boolean, double precision floating point (IEEE 754-1985) or signed 64 bit integer.
>    An array of primitive type values. The array MUST be homogeneous, i.e., it MUST NOT contain values of different types.

I don't think ValueTypeMap and ValueTypeBytes fall into these valid value types.

Reply:
How are we dealing with cases where we get invalid attribute values? Are we ignoring them?

Reply:
We had an issue in apm-server where otlp events with non-compliant attributes cause panics: https://github.com/elastic/apm-server/issues/13703

In es exporter, by looking at the code, ValueTypeMap will be successfully encoded as objects, ValueTypeBytes will be ignored. Ideally they should be dropped explicitly with a warning, but as these are non-compliant, user should expect undefined behavior, which includes successful or unsuccessful indexing.

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35023

Reply:
As long as we don't serialize non-compliant values and there's no panic, this sounds fine. Ideally, we would increment the dropped attributes count.